_resources/port1.0/group/python-1.0.tcl

```diff
@@ -270,22 +273,64 @@ proc python_set_default_version {option action args} {
 
 options python.branch python.prefix python.bin python.lib python.libdir \
         python.include python.pkgd
-# for pythonXY, python.branch is X.Y
-default python.branch   {[string range ${python.version} 0 end-1].[string index ${python.version} end]}
-default python.prefix   {${frameworks_dir}/Python.framework/Versions/${python.branch}}
-default python.bin      {${python.prefix}/bin/python${python.branch}}
-default python.lib      {${python.prefix}/Python}
-default python.pkgd     {${python.prefix}/lib/python${python.branch}/site-packages}
-default python.libdir   {${python.prefix}/lib/python${python.branch}}
+# for pythonXY or pypyXY, python.branch is X.Y
+default python.branch   {[string index ${python.version} end-1].[string index ${python.version} end]}
```

Comment:
You didn't introduce this issue, but it's worth noting that this will break with python 3.10.

Reply:
I fixed it, while at it, and now use a single regular expression for extracting the version. Everywhere else just checks for PyPy and version 2.7 — which now works.